packages/ckb/src/spore/leap.ts

```diff
@@ -0,0 +1,270 @@
+import { BtcTimeCellsParams, RgbppCkbVirtualTx } from '../types/rgbpp';
+import { append0x, calculateTransactionFee } from '../utils';
+import {
+  btcTxIdFromBtcTimeLockArgs,
+  buildSpvClientCellDep,
+  calculateCommitment,
+  compareInputs,
+  genBtcTimeLockScript,
+  lockScriptFromBtcTimeLockArgs,
+  transformSpvProof,
+} from '../utils/rgbpp';
+import {
+  Hex,
+  LeapSporeFromBtcToCkbVirtualTxParams,
+  LeapSporeFromCkbToBtcVirtualTxParams,
+  SporeLeapVirtualTxResult,
+} from '../types';
+import {
+  BTC_JUMP_CONFIRMATION_BLOCKS,
+  RGBPP_TX_WITNESS_MAX_SIZE,
+  RGBPP_WITNESS_PLACEHOLDER,
+  getBtcTimeLockConfigDep,
+  getBtcTimeLockDep,
+  getRgbppLockConfigDep,
+  getRgbppLockDep,
+  getRgbppLockScript,
+  getSporeTypeDep,
+} from '../constants';
+import { generateSporeTransferCoBuild } from '../utils/spore';
+import { NoRgbppLiveCellError, RgbppUtxoBindMultiSporesError } from '../error';
+import {
+  addressToScript,
+  getTransactionSize,
+  serializeOutPoint,
+  serializeScript,
+  serializeWitnessArgs,
+} from '@nervosnetwork/ckb-sdk-utils';
+import { blockchain } from '@ckb-lumos/base';
+import { buildBtcTimeUnlockWitness } from '../rgbpp';
+
+/**
+ * Generate the virtual ckb transaction for leaping spore from BTC to CKB
+ * @param collector The collector that collects CKB live cells and transactions
+ * @param sporeRgbppLockArgs The spore rgbpp cell lock script args whose data structure is: out_index | bitcoin_tx_id
+ * @param sporeTypeBytes The spore type script serialized bytes
+ * @param toCkbAddress The receiver ckb address
+ * @param witnessLockPlaceholderSize The WitnessArgs.lock placeholder bytes array size and the default value is 5000
+ * @param ckbFeeRate The CKB transaction fee rate, default value is 1100
+ */
+export const genLeapSporeFromBtcToCkbVirtualTx = async ({
+  collector,
+  sporeRgbppLockArgs,
+  sporeTypeBytes,
+  toCkbAddress,
+  isMainnet,
+  witnessLockPlaceholderSize,
+  ckbFeeRate,
+}: LeapSporeFromBtcToCkbVirtualTxParams): Promise<SporeLeapVirtualTxResult> => {
+  const sporeRgbppLock = {
+    ...getRgbppLockScript(isMainnet),
+    args: append0x(sporeRgbppLockArgs),
+  };
+  const sporeCells = await collector.getCells({ lock: sporeRgbppLock, isDataEmpty: false });
+  if (!sporeCells || sporeCells.length === 0) {
+    throw new NoRgbppLiveCellError('No spore rgbpp cells found with the spore rgbpp lock args');
+  }
+  if (sporeCells.length > 1) {
+    throw new RgbppUtxoBindMultiSporesError('The UTXO is bound to multiple spores');
```

Comment:
```suggestion
    throw new RgbppUtxoBindMultiSporesError('The BTC UTXO is bound to multiple CKB cells');
```

Reply:
To avoid ambiguity I updated the error message as blow:
```
The BTC UTXO must not be bound to multiple CKB cells
```